### PR TITLE
Ralph2 sync using hermes

### PR DESCRIPTION
**Pull request by mkurek**

### CODE REVIEW

File: src/ralph/ralph2_sync/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+default_app_config = 'ralph.ralph2_sync.apps.Ralph2SyncConfig'
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/__init__.py]: \[Pylint\] INFO: Missing module docstring \[missing\-docstring\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/__init__.py]: \[Pylint\] INFO: Invalid constant name "default\_app\_config" \[invalid\-name\]

File: src/ralph/ralph2_sync/apps.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from ralph.apps import RalphAppConfig
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/apps.py]: \[Pylint\] INFO: Missing module docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/apps.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+
+class Ralph2SyncConfig(RalphAppConfig):
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/apps.py]: \[Pylint\] INFO: Missing class docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/publishers.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        )
+        # register publisher
+        @pyhermes.publisher(
+            topic=func.__name__
+        )
+        def wrapped_func(*args, **kwargs):
+            # publish only if sync enabled (globally and for particular
+            # function)
+            if (
+                settings.RALPH2_HERMES_SYNC_ENABLED and
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
                settings\.RALPH2\_HERMES\_SYNC\_ENABLED and
                ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                func.__name__ in settings.RALPH2_HERMES_SYNC_FUNCTIONS
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
                func\.\_\_name\_\_ in settings\.RALPH2\_HERMES\_SYNC\_FUNCTIONS
                ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        # location
+        'data_center': _get_obj_id_ralph_20(
+            asset.rack.server_room.data_center
+        ),
+        'server_room': _get_obj_id_ralph_20(asset.rack.server_room),
+        'rack': _get_obj_id_ralph_20(asset.rack),
+    }
+    # simple fields
+    for field in [
+        'id', 'orientation', 'position', 'sn', 'barcode', 'slot_no',
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'id', 'orientation', 'position', 'sn', 'barcode', 'slot\_no',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+        'price', 'niw', 'task_url', 'remarks', 'order_no', 'invoice_date',
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'price', 'niw', 'task\_url', 'remarks', 'order\_no', 'invoice\_date',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        'invoice_no', 'provider', 'source', 'status', 'depreciation_rate',
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'invoice\_no', 'provider', 'source', 'status', 'depreciation\_rate',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+        'depreciation_end_date', 'management_ip', 'management_hostname',
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'depreciation\_end\_date', 'management\_ip', 'management\_hostname',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+        'hostname'
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'hostname'
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+    ]:
+        data[field] = str(getattr(asset, field, '') or '')
+    # foreign key fields
+    for field in [
+        'model', 'property_of',
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'model', 'property\_of',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py [hunk 12]
```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Missing module docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/publishers.py [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+from ralph.data_importer.models import ImportedObjects
+
+
+def ralph2_sync(model):
+    """
+    Decorator for synchronizers with Ralph2. Decorated function should return
+    dict with event data. Decorated function name is used as a topic name and
+    dispatch_uid for post_save signal.
+    """
+    def wrap(func):
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Missing function docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/publishers.py [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+        @wraps(func)
+        # connect to post_save signal for a model
+        @receiver(
+            post_save, sender=model, dispatch_uid=func.__name__,
+        )
+        # register publisher
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Missing function docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/publishers.py [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+    return wrap
+
+
+def _get_obj_id_ralph_20(obj):
+    """
+    Returns ID of object in Ralph2 or None if not found.
+    """
+    if not obj:
+        return None
+    pk = None
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Invalid variable name "pk" \[invalid\-name\]

File: src/ralph/ralph2_sync/publishers.py [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+    content_type = ContentType.objects.get_for_model(obj._meta.model)
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Access to a protected member \_meta of a client class \[protected\-access\]

File: src/ralph/ralph2_sync/publishers.py [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+    try:
+        imported_obj = ImportedObjects.objects.get(
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'objects' member \[no\-member\]

File: src/ralph/ralph2_sync/publishers.py [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+            object_pk=obj.pk,
+            content_type=content_type,
+        )
+        pk = imported_obj.old_object_pk
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Invalid variable name "pk" \[invalid\-name\]

File: src/ralph/ralph2_sync/publishers.py [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+    except (obj._meta.model.DoesNotExist, ImportedObjects.DoesNotExist):
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Access to a protected member \_meta of a client class \[protected\-access\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'DoesNotExist' member \[no\-member\]

File: src/ralph/ralph2_sync/publishers.py [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+        pass
+    return pk
+
+
+@ralph2_sync(DataCenterAsset)
+def sync_dc_asset_to_ralph2(sender, instance=None, created=False, **kwargs):
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'kwargs' \[unused\-argument\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'sender' \[unused\-argument\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'created' \[unused\-argument\]

File: src/ralph/ralph2_sync/publishers.py [hunk 21]
```diff
@@ -0,0 +1,123 @@
...
+    # foreign key fields
+    for field in [
+        'model', 'property_of',
+    ]:
+        data[field] = _get_obj_id_ralph_20(getattr(asset, field, None))
+    return data
+
+
+@ralph2_sync(AssetModel)
+def sync_model_to_ralph2(sender, instance=None, created=False, **kwargs):
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'created' \[unused\-argument\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'kwargs' \[unused\-argument\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'sender' \[unused\-argument\]

File: src/ralph/ralph2_sync/subscribers.py [hunk 22]
```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] INFO: Missing module docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/subscribers.py [hunk 23]
```diff
@@ -0,0 +1,43 @@
...
+import logging
+import pyhermes
+
+from django.contrib.contenttypes.models import ContentType
+
+from ralph.assets.models import AssetModel
+from ralph.data_center.models import DataCenterAsset
+from ralph.data_importer.models import ImportedObjects
+
+logger = logging.getLogger(__name__)
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] INFO: Invalid constant name "logger" \[invalid\-name\]

File: src/ralph/ralph2_sync/subscribers.py [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+
+model_mapping = {
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] INFO: Invalid constant name "model\_mapping" \[invalid\-name\]

File: src/ralph/ralph2_sync/subscribers.py [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+
+
+@pyhermes.subscriber(topic='ralph2_sync_ack')
+def ralph2_sync_ack(data):
+    """
+    Receives ACK from Ralph2 and checks if there is ImportedObject entry for
+    synced objects. Creates new ImportedObject entry if not exists before.
+    """
+    model = model_mapping[data['model']]
+    ct = ContentType.objects.get_for_model(model)
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] INFO: Invalid variable name "ct" \[invalid\-name\]

File: src/ralph/ralph2_sync/subscribers.py [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+    try:
+        ImportedObjects.objects.get(
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'objects' member \[no\-member\]

File: src/ralph/ralph2_sync/subscribers.py [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+            content_type=ct,
+            object_pk=data['ralph3_id']
+        )
+        logger.info(
+            'ImportedObject mapping for {} found in Ralph3'.format(data)
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] WARNING: Use % formatting in logging functions and pass the % parameters as arguments \[logging\-format\-interpolation\]

File: src/ralph/ralph2_sync/subscribers.py [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+        )
+    except ImportedObjects.DoesNotExist:
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'DoesNotExist' member \[no\-member\]

File: src/ralph/ralph2_sync/subscribers.py [hunk 29]
```diff
@@ -0,0 +1,43 @@
...
+        logger.info(
+            'Creating new ImportedObject mapping in Ralph3: {}'.format(data)
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] WARNING: Use % formatting in logging functions and pass the % parameters as arguments \[logging\-format\-interpolation\]

File: src/ralph/ralph2_sync/subscribers.py [hunk 30]
```diff
@@ -0,0 +1,43 @@
...
+        )
+        ImportedObjects.objects.create(
```
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'objects' member \[no\-member\]

File: src/ralph/ralph2_sync/__init__.py (hunk 1 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/__init__.py]: \[Pylint\] INFO: Missing module docstring \[missing\-docstring\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/__init__.py]: \[Pylint\] INFO: Invalid constant name "default\_app\_config" \[invalid\-name\]

File: src/ralph/ralph2_sync/apps.py (hunk 2 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/apps.py]: \[Pylint\] INFO: Missing module docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/apps.py (hunk 3 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/apps.py]: \[Pylint\] INFO: Missing class docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/publishers.py (hunk 4 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
                settings\.RALPH2\_HERMES\_SYNC\_ENABLED and
                ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py (hunk 5 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
                func\.\_\_name\_\_ in settings\.RALPH2\_HERMES\_SYNC\_FUNCTIONS
                ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py (hunk 6 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'id', 'orientation', 'position', 'sn', 'barcode', 'slot\_no',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py (hunk 7 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'price', 'niw', 'task\_url', 'remarks', 'order\_no', 'invoice\_date',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py (hunk 8 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'invoice\_no', 'provider', 'source', 'status', 'depreciation\_rate',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py (hunk 9 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'depreciation\_end\_date', 'management\_ip', 'management\_hostname',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py (hunk 10 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'hostname'
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py (hunk 11 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Wrong hanging indentation before block \(add 4 spaces\)\.
        'model', 'property\_of',
        ^   | \[bad\-continuation\]

File: src/ralph/ralph2_sync/publishers.py (hunk 12 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Missing module docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/publishers.py (hunk 13 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Missing function docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/publishers.py (hunk 14 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Missing function docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/publishers.py (hunk 15 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Invalid variable name "pk" \[invalid\-name\]

File: src/ralph/ralph2_sync/publishers.py (hunk 16 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Access to a protected member \_meta of a client class \[protected\-access\]

File: src/ralph/ralph2_sync/publishers.py (hunk 17 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'objects' member \[no\-member\]

File: src/ralph/ralph2_sync/publishers.py (hunk 18 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] INFO: Invalid variable name "pk" \[invalid\-name\]

File: src/ralph/ralph2_sync/publishers.py (hunk 19 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Access to a protected member \_meta of a client class \[protected\-access\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'DoesNotExist' member \[no\-member\]

File: src/ralph/ralph2_sync/publishers.py (hunk 20 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'kwargs' \[unused\-argument\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'created' \[unused\-argument\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'sender' \[unused\-argument\]

File: src/ralph/ralph2_sync/publishers.py (hunk 21 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'created' \[unused\-argument\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'sender' \[unused\-argument\]

Comment (sputnikci) [inline src/ralph/ralph2_sync/publishers.py]: \[Pylint\] WARNING: Unused argument 'kwargs' \[unused\-argument\]

File: src/ralph/ralph2_sync/subscribers.py (hunk 22 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] INFO: Missing module docstring \[missing\-docstring\]

File: src/ralph/ralph2_sync/subscribers.py (hunk 23 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] INFO: Invalid constant name "logger" \[invalid\-name\]

File: src/ralph/ralph2_sync/subscribers.py (hunk 24 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] INFO: Invalid constant name "model\_mapping" \[invalid\-name\]

File: src/ralph/ralph2_sync/subscribers.py (hunk 25 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] INFO: Invalid variable name "ct" \[invalid\-name\]

File: src/ralph/ralph2_sync/subscribers.py (hunk 26 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'objects' member \[no\-member\]

File: src/ralph/ralph2_sync/subscribers.py (hunk 27 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] WARNING: Use % formatting in logging functions and pass the % parameters as arguments \[logging\-format\-interpolation\]

File: src/ralph/ralph2_sync/subscribers.py (hunk 28 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'DoesNotExist' member \[no\-member\]

File: src/ralph/ralph2_sync/subscribers.py (hunk 29 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] WARNING: Use % formatting in logging functions and pass the % parameters as arguments \[logging\-format\-interpolation\]

File: src/ralph/ralph2_sync/subscribers.py (hunk 30 above)
Comment (sputnikci) [inline src/ralph/ralph2_sync/subscribers.py]: \[Pylint\] ERROR: Class 'ImportedObjects' has no 'objects' member \[no\-member\]

**Comment by ar4s** [comment]
:ok:

File: src/ralph/ralph2_sync/publishers.py [hunk 31]
```diff
@@ -0,0 +1,134 @@
...
+                    pyhermes.publish(func.__name__, result)
+                except:
+                    logger.exception('Error during Ralph2 sync')
+                else:
+                    return result
+        return wrapped_func
+    return wrap
+
+
+def _get_obj_id_ralph_20(obj):
```
Comment (ghost) [inline src/ralph/ralph2_sync/publishers.py]: Isn't this duplicated? I think something similar is implemented in dataimporter.

File: src/ralph/ralph2_sync/subscribers.py [hunk 32]
```diff
@@ -0,0 +1,86 @@
...
+    Log additional exception when sync has failed.
+    """
+    def _get_wrapper(self, func):
+        @wraps(func)
+        @transaction.atomic
+        def exception_wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except:
+                logger.exception('Exception during syncing')
```
Comment (ghost) [inline src/ralph/ralph2_sync/subscribers.py:32]: Should we log here other important information?
I see here only: `'Exception during syncing'`

File: src/ralph/ralph2_sync/publishers.py [hunk 33]
```diff
@@ -0,0 +1,134 @@
...
+    """
+    Returns ID of object in Ralph2 or None if not found.
+    """
+    if not obj:
+        return None
+    pk = None
+    content_type = ContentType.objects.get_for_model(obj._meta.model)
+    try:
```
Comment (szok) [inline src/ralph/ralph2_sync/publishers.py:63]: Maybe use: ImportedObjects .get_imported_id(obj) ?
https://github.com/allegro/ralph/blob/ng/src/ralph/data_importer/models.py#L65

File: src/ralph/ralph2_sync/publishers.py (hunk 31 above)
Comment (szok) [inline src/ralph/ralph2_sync/publishers.py]: why 20?

File: src/ralph/ralph2_sync/subscribers.py [hunk 34]
```diff
@@ -0,0 +1,86 @@
...
+            object_pk=data['ralph3_id']
+        )
+        logger.info(
+            'ImportedObject mapping for {} found in Ralph3'.format(data)
+        )
+    except ImportedObjects.DoesNotExist:
+        logger.info(
+            'Creating new ImportedObject mapping in Ralph3: {}'.format(data)
+        )
+        ImportedObjects.objects.create(
```
Comment (szok) [inline src/ralph/ralph2_sync/subscribers.py:56]: Maybe use: ImportedObject.create() ?

File: src/ralph/settings/base.py [hunk 35]
```diff
@@ -346,3 +347,7 @@ def os_env_true(var, default=''):
     INSTALLED_APPS += (
         'pyhermes.apps.django',
     )
+
+
+RALPH2_HERMES_SYNC_ENABLED = False
```
Comment (szok) [inline src/ralph/settings/base.py]: why not use enviroments?

File: src/ralph/ralph2_sync/publishers.py (hunk 31 above)
Comment (mkurek) [inline src/ralph/ralph2_sync/publishers.py]: @xliiv no - we don't have any function like this right now (we have function `get_imported_obj` which returns new object based on model and old object id). @szok changed to 2

File: src/ralph/ralph2_sync/subscribers.py (hunk 32 above)
Comment (mkurek) [inline src/ralph/ralph2_sync/subscribers.py:32]: `logger.exception` will log traceback by default (our msg is just additional one to the traceback)

File: src/ralph/ralph2_sync/subscribers.py (hunk 34 above)
Comment (mkurek) [inline src/ralph/ralph2_sync/subscribers.py:56]: I don't have new object - just its PK.

**Comment by ghost** [comment]
:ok:

**Comment by szok** [comment]
:ok:

**Pull request merged**